src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GenericTypesTemplateMap.cs

```diff
@@ -136,8 +122,11 @@ public static bool IsArrayTypeEligibleForTemplate(TypeDesc type)
             return !elementType.IsPointer && !elementType.IsFunctionPointer;
         }
 
-        public static bool IsEligibleToHaveATemplate(TypeDesc type)
+        public static bool IsEligibleToHaveATemplate(NodeFactory factory, TypeDesc type)
         {
+            if (type.HasSameTypeDefinition(factory.ArrayOfTClass))
```

Comment:
Yeah thinking about it again this is probably going to result in a missing template in some weird case. Commenting it out causes Array<T> EETypes to be rooted again. Ugh - I'll take a look.